sklearn/model_selection/tests/test_split.py

```diff
@@ -804,6 +806,69 @@ def test_leave_one_p_group_out_error_on_fewer_number_of_groups():
                          LeavePGroupsOut(n_groups=3).split(X, y, groups))
 
 
+def test_repeated_kfold_value_errors():
+    # n_repeats is not integer or <= 1
+    assert_raises(ValueError, RepeatedKFold, n_repeats=1)
+    assert_raises(ValueError, RepeatedKFold, n_repeats=1.5)
+
+
+def test_repeated_kfold_determinstic_split():
+    X = [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]]
+    random_state = 258173307
+    splits = RepeatedKFold(
+        n_splits=2,
+        n_repeats=2,
+        random_state=random_state).split(X)
+
+    train, test = next(splits)
+    assert_array_equal(train, [0, 3])
+    assert_array_equal(test, [1, 2, 4])
+
+    train, test = next(splits)
+    assert_array_equal(train, [1, 2, 4])
+    assert_array_equal(test, [0, 3])
+
+    train, test = next(splits)
+    assert_array_equal(train, [3, 4])
+    assert_array_equal(test, [0, 1, 2])
+
+    train, test = next(splits)
+    assert_array_equal(train, [0, 1, 2])
+    assert_array_equal(test, [3, 4])
+
```

Comment:
add a comment
perhaps use a loop or a helper function to avoid duplicated code.